pkg/storage/backend-storage/backend-storage.go

```diff
@@ -104,6 +111,119 @@ func PVCForMigrationTarget(pvcStore cache.Store, migration *corev1.VirtualMachin
 	return pvcForMigrationTargetFromStore(pvcStore, migration)
 }
 
+func RecoverFromBrokenMigration(client kubecli.KubevirtClient, migrationIndexer cache.Indexer, pvcStore cache.Store, vmi *corev1.VirtualMachineInstance, launcherImage string) (*v1.PersistentVolumeClaim, error, bool) {
+	var pvc *v1.PersistentVolumeClaim
+
+	migration, err := migrations.MigrationForVMI(migrationIndexer, vmi)
+	if err != nil || migration == nil {
+		return nil, err, false
+	}
+	if migration.Status.MigrationState == nil ||
+		migration.Status.MigrationState.TargetPersistentStatePVCName == migration.Status.MigrationState.SourcePersistentStatePVCName {
+		// The migration either didn't actually start, or the backend storage is RWX. Either way we're good, we can delete it.
+		err := client.VirtualMachineInstanceMigration(migration.Namespace).Delete(context.Background(), migration.Name, metav1.DeleteOptions{})
+		return nil, err, false
+	}
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "backend-storage-recover-",
+		},
+		Spec: batchv1.JobSpec{
+			ActiveDeadlineSeconds:   pointer.P(int64(30)),
+			BackoffLimit:            pointer.P(int32(1)),
+			TTLSecondsAfterFinished: pointer.P(int32(30)),
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "backend-storage-recover-",
+				},
+				Spec: v1.PodSpec{
+					RestartPolicy: v1.RestartPolicyNever,
+					SecurityContext: &v1.PodSecurityContext{
+						RunAsNonRoot: pointer.P(true),
+						RunAsUser:    pointer.P(int64(util.NonRootUID)),
+						RunAsGroup:   pointer.P(int64(util.NonRootUID)),
+						FSGroup:      pointer.P(int64(util.NonRootUID)),
+						SeccompProfile: &v1.SeccompProfile{
+							Type: v1.SeccompProfileTypeRuntimeDefault,
+						},
+					},
+					Containers: []v1.Container{{
+						Name: "container",
+						SecurityContext: &v1.SecurityContext{
+							AllowPrivilegeEscalation: pointer.P(false),
+							Capabilities:             &v1.Capabilities{Drop: []v1.Capability{"ALL"}},
+						},
+						Image:   launcherImage,
+						Command: []string{"ls"},
+						Args:    []string{"/meta/migrated"},
```

Comment:
What does this file represents?

Reply:
This file is created by the libvirt hook and tells us the VM was successfully migrated.
The point of this whole PR (and I should make that clearer) is to address the case where libvirt finished a migration but the info didn't make it to our migration controller for any reason.

Reply:
Can we add this as a comment somewhere?